util/arbmath/math.go

```diff
@@ -411,15 +413,25 @@ func ApproxExpBasisPoints(value Bips, accuracy uint64) Bips {
 	// This cast is safe because input is always positive
 	// #nosec G115
 	x := uint64(input)
-	bips := uint64(OneInBips)
-
-	res := bips + x/accuracy
-	for i := uint64(1); i < accuracy; i++ {
-		res = bips + SaturatingUMul(res, x)/((accuracy-i)*bips)
+	one := uint64(OneInBips)
+
+	// This is actually an optimization for computing the Maclaurin series
+	// inspired by Horner's method for solving taylor polynomials.
+	// It is used to reduce the number of multiplications and divisions required
+	// to compute the polynomial series.
+	// However, this is not a "pure" implementation of Horner's method, as it
+	// takes a short-cut and initializes the result with the nth term of
+	// the series before iterating through the rest of the terms.
+	// In Horner's method, the nth term would not normally be included in the
+	// result, but this is an optimization to improve accuracy for small values
+	// of accuracy.
+	res := one + x/accuracy
```

Comment:
This is the line that I cannot make sense of.

one and x are both values that have been multiplied by 10000 but, accuracy hasn't been.

For example, if we wanted to calculate e^0.05 we would call this method like:
`ApproxExpBasisPoints(500, 4)`

And, in each iteration of the for loop, the divisor is multiplied by 10000.
But, out here before we go into the for loop, we're not multiplying accuracy by 10000.

I really think this is a "bug" or at least a happy accident. But, the problem is, if we fix it, it changes the State-Transition function, and the replay.wasm wouldn't work with older blocks.

Reply:
FYI, it's been that way since the method was introduced 3 years ago:
https://github.com/OffchainLabs/nitro/blame/b69449e41ff76bab1f6ac290dd8989f503226c7c/util/math.go#L191

Reply:
yes, if you are going to do the loop forward then you need to start with `one + x / (accuracy * one)` but I believe that loop would be more expensive than the original one that has a clever trick to avoid one multiplication. The original one does not need to start with this extra multiplication, explained in the next comment